test/k8sT/Services.go

```diff
@@ -832,14 +832,16 @@ var _ = Describe("K8sServicesTest", func() {
 				}
 			}
 
+			By("Removing %s pod so that another pod is chosen", dstPod)
+
 			// Delete the pod, and check that a new backend is chosen
-			nodes, err := kubectl.GetPodsNodes(helpers.DefaultNamespace, dstPod)
-			Expect(err).Should(BeNil(), "Cannot get node name of %s pod", dstPod)
 			kubectl.DeleteResource("pod", dstPod).ExpectSuccess("Unable to delete %s pod", dstPod)
-			// Wait until the pod has been removed from the cilium endpoints list.
-			// Otherwise, the requests below might fail as the non-existing endpoint
-			// will be chosen.
-			kubectl.WaitForCiliumEndpointDeleted(nodes[dstPod], helpers.DefaultNamespace, dstPod)
+
+			// Wait until the replacement pod has been provisioned + 1 second. This
+			// should give enough time for cilium-agents to remove the deleted pod
+			// from the BPF LB maps, so that the next request won't choose the deleted pod.
+			waitPodsDs()
+			time.Sleep(1 * time.Second)
```

Comment:
Do you think that waiting 1 second is reliable?

I've often seen `kubectl delete` commands hang for some time, more than one second. I don't have a direct comparison with this as I didn't dig into what `waitPodsDs()` does here though.

Reply:
Apparently not. Did more troubleshooting and found out that ipcache entry is missing for the new backend after the removal which resulted in the request being drop.